packages/aws-cdk/lib/api/hotswap/lambda-functions.ts

```diff
@@ -369,38 +346,44 @@ async function versionsAndAliases(logicalId: string, evaluateCfnTemplate: Evalua
   // find all Lambda Aliases that reference the above Versions
   const aliasesReferencingVersions = flatMap(versionsReferencingFunction, v =>
     evaluateCfnTemplate.findReferencesTo(v.LogicalId));
-  // Limited set of updates per function
-  // eslint-disable-next-line @cdklabs/promiseall-no-unbounded-parallelism
-  const aliasesNames = await Promise.all(aliasesReferencingVersions.map(a =>
-    evaluateCfnTemplate.evaluateCfnExpression(a.Properties?.Name)));
 
-  return { versionsReferencingFunction, aliasesNames };
+  return { versionsReferencingFunction, aliasesReferencingVersions };
 }
 
-/**
- * Renders the string used in displaying Alias resource names that reference the specified Lambda Function
- */
-async function renderAliases(
+async function dependantResources(
   logicalId: string,
+  functionName: string,
   evaluateCfnTemplate: EvaluateCloudFormationTemplate,
-  callbackfn: (value: any, index: number, array: any[]) => Promise<string>,
-): Promise<string[]> {
-  const aliasesNames = (await versionsAndAliases(logicalId, evaluateCfnTemplate)).aliasesNames;
+): Promise<Array<{
+    logicalId: string;
+    resourceType: string;
+    physicalName?: string;
+    description: string;
```

Comment:
💅 This would be nicer with a named struct definition.

Reply:
Coming up next!